### PR TITLE
SNOW-1061854 Run e2e tests with single buffer

### PR DESCRIPTION
**Pull request by sfc-gh-lshcharbaty**
# Overview

SNOW-1061854

Running e2e tests with single and double buffers. In every snowpipe_streaming test we'll need to specify `$SNOWFLAKE_STREAMING_ENABLE_SINGLE_BUFFER` property that's later substituted with `true` and `false` values. To achieve isolation between different parameter sets, a parameter set index is added to test salt.

I wouldn't use github test matrix, as this solution seems to be temporary and removed shortly.

**Note that currently the tests are run sequentially that will take about 2x time!** File generation logic needs to be rewritten to allow running a test suite with different parameters in parallel.

All related changes are in [test_verfiy.py](https://github.com/snowflakedb/snowflake-kafka-connector/pull/867/files#diff-0eb8c275e2e077d9fe6b7d9a9e45cd7887d93c752cb0013ccf646fb41e90de59). 
Also removed a check for enabled `SNOWPIPE_STREAMING_ENABLE_SINGLE_BUFFER` in [Utils.java](https://github.com/snowflakedb/snowflake-kafka-connector/pull/867/files#diff-371eabc61cd3788bddba04e3204eaa5ad54d315769e30def9b88095307e5a4b1).
The rest are json request and test verification adjustments.

A few e2e tests don't pass with a single buffer. See SNOW-1061855 and SNOW-1475664 for the details.

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [x] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

File: test/rest_request_template/travis_correct_snowpipe_streaming_string_json.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "snowflake.schema.name": "SNOWFLAKE_SCHEMA",
     "snowflake.role.name": "SNOWFLAKE_ROLE",
     "snowflake.ingestion.method": "SNOWPIPE_STREAMING",
+    "snowflake.streaming.enable.single.buffer": "$SNOWFLAKE_STREAMING_ENABLE_SINGLE_BUFFER",
```
Comment (sfc-gh-lshcharbaty) [inline test/rest_request_template/travis_correct_snowpipe_streaming_string_json.json:17]: This line is going to be added for each `SNOWPIPE_STREAMING` test.

**Review by sfc-gh-krosinski — APPROVED** [review]

File: test/rest_request_template/travis_correct_snowpipe_streaming_string_json.json (hunk 1 above)
Comment (sfc-gh-achyzy) [reply test/rest_request_template/travis_correct_snowpipe_streaming_string_json.json:17]: Why introduce a new way of introducing parameters with '$'?

File: test/test_verify.py [hunk 2]
```diff
@@ -275,6 +295,9 @@ def cleanTableStagePipe(self, connectorName, topicName="", partitionNumber=1):
             print(datetime.now().strftime("%H:%M:%S "), "=== Drop pipe {} ===".format(pipeName))
             self.snowflake_conn.cursor().execute("DROP pipe IF EXISTS {}".format(pipeName))
 
+        print(datetime.now().strftime("%H:%M:%S "), f"=== Delete Kafka topic {topicName} ===")
+        self.deleteTopic(topicName)
```
Comment (sfc-gh-achyzy) [inline test/test_verify.py:299]: Why exactly do we need to delete the topic here? Is this a fix, as it was missing, or did you add some logic related to that?

File: test/rest_request_template/travis_correct_snowpipe_streaming_string_json.json (hunk 1 above)
Comment (sfc-gh-lshcharbaty) [reply test/rest_request_template/travis_correct_snowpipe_streaming_string_json.json:17]: Using caps only doesn't make the template parameters stand out. 
In this example `SNOWFLAKE_SCHEMA` and `SNOWFLAKE_ROLE` are parameter names while `SNOWPIPE_STREAMING` is a value.

I'm also inclined to adjust the naming for all template parameters.

File: test/run_test_apache.sh [hunk 3]
```diff
@@ -175,6 +175,7 @@ if [ $testError -ne 0 ]; then
     RED='\033[0;31m'
     NC='\033[0m' # No Color
     echo -e "${RED} There is error above this line ${NC}"
-    cat $APACHE_LOG_PATH/kc.log
+    #    Ignore for e2e test draft. Todo: turn back again?
+    #    cat $APACHE_LOG_PATH/kc.log
```
Comment (sfc-gh-lshcharbaty) [inline test/run_test_apache.sh:179]: It's horrible to check what has failed when printing **all** Kafka connector debug logs. Once all tests have passed, this change will be reverted back.

**Review by sfc-gh-achyzy — APPROVED** [review]
LGTM. I hope you'll be able to get to the state where all merge gates pass.

File: test/test_verify.py [hunk 4]
```diff
@@ -18,14 +20,31 @@
...
+class ConnectorParameters:
+    snowflake_streaming_enable_single_buffer: str
+
+
+class ConnectorParametersList:
+    def __init__(self, connectorParametersList: list[ConnectorParameters]):
+        self.connectorParametersList = connectorParametersList
+
+    def for_each(self, function: Callable[[int, ConnectorParameters], None]) -> None:
+        for idx, conenctor_parameters in enumerate(self.connectorParametersList):
```
Comment (sfc-gh-akowalczyk) [inline test/test_verify.py:33]: typo, `connector_parameters`,

shouldn't it be `connector_parameter`, BTW?

**Review by sfc-gh-akowalczyk — APPROVED** [review]
Good idea to set the parameter at the test suite top level

**Pull request merged**